sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,654 @@
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_result_t dif_pinmux_init(dif_pinmux_params_t params,
+                                    dif_pinmux_t *pinmux);
+
+/**
+ * Locks out Pin Multiplexer functionality based on the `target` and `index`.
+ *
+ * This function allows for a fine grained locking of the Pin Multiplexer
+ * parts. `index` changes meaning dependent on the `target`. For example, when
+ * `target` is `kDifPinmuxLockTargetInsel`, `index` represents a peripheral
+ * input for which the MIO input select functionality should be locked.
+ *
+ * NOTE:
+ * Locking an already locked register will succeed with the `kDifPinmuxOk`
+ * return code.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param index Exact meaning is determined by the `target`, however, it
+ *              represents one of the following: peripheral input index,
+ *              MIO pad index, DIO pad index or a wakeup detector index.
+ * @param target Target functionality to be locked.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_result_t dif_pinmux_lock(dif_pinmux_t *pinmux,
+                                    dif_pinmux_index_t index,
+                                    dif_pinmux_lock_target_t target);
+
+/**
+ * Obtains the Pin Multiplexer Pad lock status based on `target` and `index`.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param index Exact meaning is determined by the `target`, however, it
+ *              represents one of the following: peripheral input index,
+ *              MIO pad index, DIO pad index or a wakeup detector index.
+ * @param target Target functionality to be locked.
+ * @param[out] is_locked Out-param for the locked state.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_result_t dif_pinmux_is_locked(const dif_pinmux_t *pinmux,
+                                         dif_pinmux_index_t index,
+                                         dif_pinmux_lock_target_t target,
+                                         bool *is_locked);
+
+/**
+ * The result of a Pin Multiplexer input or output select operations.
+ */
+typedef enum dif_pinmux_select_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifPinmuxSelectOk = kDifPinmuxOk,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifPinmuxSelectError = kDifPinmuxError,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occurred.
+   */
+  kDifPinmuxSelectBadArg = kDifPinmuxBadArg,
+
+  /**
+   * Indicates that the operation is not permitted (locked).
+   */
+  kDifPinmuxSelectLocked,
+} dif_pinmux_select_result_t;
+
+/**
+ * Sets a connection between a peripheral input and a MIO pad input.
+ *
+ * A peripheral input can be connected to any available MIO pad input.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param input A Peripheral input index.
```

Comment:
This name `input` doesn't match the name in the function prototype (`peripheral_input`).

Reply:
Good catch.

Done.